src/pixelator/pna/plot/__init__.py

```diff
@@ -0,0 +1,60 @@
+"""Functions for creating plots that are useful with PNA data.
+
+Copyright © 2025 Pixelgen Technologies AB.
+"""
+
+from typing import Optional, Tuple
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+
+def molecule_rank_plot(
+    data: pd.DataFrame, group_by: Optional[str] = None
+) -> Tuple[plt.Figure, plt.Axes]:
+    """Plot the number of molecules (n_umi) per component against its n_umi rank.
+
+    :param data: a pandas DataFrame with a column 'n_umi' containing antibody
+    counts in components.
+    :param group_by: a column in the DataFrame to group the plot by.
+
+    :return: a plot showing the number of molecules per component against its
+    molecule rank used for quality control.
+    :rtype: Tuple[plt.Figure, plt.Axes]
+    :raises: AssertionError if the required column(s) are not present in the DataFrame
+    :raises: ValueError if the data types are invalid
+    """
```

Comment:
We are gradually trying to go over to the google style doc strings, so perhaps we should update that here now that we are adding a new function?